### PR TITLE
Allow COG selection for interactions

### LINKED ISSUE #5422: Add special `Consolidated` COG Interactions behavior
When collection transactions involve a Collection Object that is part of a COG—such as loans, accessions, deaccessions, gifts for determination, or relocation for exhibition—Specify will identify and present information on other cataloged Collection Objects belonging to that COG.

If the other Collection Objects are physically attached to the Collection Object being transacted, they will also be included in the management transaction. This feature allows the user to update the transaction status of all Collection Objects within the COG.

* This logic is only necessary when the COG Type is `Consolidated`; it does not apply in other cases, such as when it is `Discrete` or `Drill Core`.

---

### User Interface Components

Record sets of COGs should be usable as the basis for a transaction just as record sets of COs are. 

Users should also be able to enter a list of delimiter-separated COG names to guide their interactions:

image

All sections expanded:
image

Once entered, all loanable preparations associated with COs in that COG should be displayed in the following menu.

### COG Identification and Display

-  A new column should be added to the "Preparations" dialog to display the formatted table record preview for the COG, in addition to Cat #, Taxon, Object Type, Select, Available, and Unavailable.
- If there is not a formatter defined for the COG table, use the ID from the database

### Error Handling and Validation

-  Prevent proceeding with incomplete transactions (e.g., preparations must be loaned together as they are consolidated).

### Returning Loans

- The "Return Loan" dialog should also display a column with the formatted COG record in it for reference
- When one preparation from the COG is selected for return, other preparations in that COG should be automatically selected as well

### PR DESCRIPTION
**Pull request by CarolineDenis**
fixes #5422

### Checklist

- [ ] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add automated tests
- [ ] Add relevant issue to release milestone

### Testing instructions

For the preparation:
For all COs be sure they have preps
- Create RS (RS COs with 1 conso child) with different COs, have at least one of them be a part of a consolidate COG 
- Create RS (RS COs with 0 conso child) with different COs, be sure no COs are part of a conso COG 
- Create RS (RS COGs with 1 conso COG) with different COGs, have at least one of them be a consolidate COG 
- Create RS (RS COGs with 0 conso COG) with different COGs, be sure no COGs are conso COG
- Create a mix RS 

For testing: 

CO not Conso COG child using enter cat num
- Open Loan dialog 
- Enter the cat num of a CO that is NOT a part of conso COG 
- Add a prep to the loan 
- [ ] Verify that only the prep you selected has been added to the Loan perps 

CO Conso COG child using enter cat num
- Open Loan dialog 
- Enter the cat num of a CO that is a part of conso COG 
- Add a prep to the loan 
- [ ] Verify that the prep you selected has been added to the Loan perps + the siblings of that CO

CO RS using choose RS
- Open Loan dialog 
- Choose RS with COs
- Add a prep to the loan 
- [ ] Verify that only the prep you selected has been added to the Loan perps

COG non conso using choose RS
- Open Loan dialog 
- Choose RS with COGs that are not conso
- Add a prep to the loan 
- [ ] Verify that only the prep you selected has been added to the Loan perps

COG conso using choose RS
- Open Loan dialog 
- Choose RS with COGs that are conso
- Add a prep to the loan 
- [ ] Verify that the prep you selected has been added to the Loan perps + all the siblings

COG mix using choose RS
- Open Loan dialog 
- Choose RS with mix COGs
- Add a prep to the loan from the conso COG and a prep from the other type
- [ ] Verify that only the prep you selected from the non conso has been added to the Loan perps + the prep from the conso COG and all its siblings

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/Interactions/PrepDialog.tsx [hunk 1]
```diff
@@ -171,22 +190,46 @@ export function PrepDialog({
...
+
+          void fetchSiblings(preparationIds)
+            .then((siblings) => {
+              const siblingsPreps = siblings.map((preparation) => {
+                const result = new itemTable.Resource();
+                result.set(
+                  'preparation',
+                  getResourceApiUrl('Preparation', preparation)
+                );
+                // Need to find a way to set the maximum
```
Comment (CarolineDenis) [inline specifyweb/frontend/js_src/lib/components/Interactions/PrepDialog.tsx:205]: todo

**Review by emenslin — DISMISSED** [review]
- [x] Verify that only the prep you selected has been added to the Loan perps 
- [x] Verify that the prep you selected has been added to the Loan perps + the siblings of that CO
- [x] Verify that only the prep you selected has been added to the Loan perps
- [ ] Verify that only the prep you selected has been added to the Loan perps
- [ ] Verify that the prep you selected has been added to the Loan perps + all the siblings
- [ ] Verify that only the prep you selected from the non conso has been added to the Loan perps + the prep from the conso COG and all its siblings

Trying to add a COG rs to a loan throws an error. 

https://github.com/user-attachments/assets/6e7e5688-5c28-4a28-a7ad-91457abd0296


[Specify 7 Crash Report - 2024-12-17T16_07_07.052Z.txt](https://github.com/user-attachments/files/18168372/Specify.7.Crash.Report.-.2024-12-17T16_07_07.052Z.txt)

**Review by alesan99 — APPROVED** [review]
### Testing instructions

- [x] Verify that only the prep you selected has been added to the Loan perps 
- [x] Verify that the prep you selected has been added to the Loan perps + the siblings of that CO
- [x] Verify that only the prep you selected has been added to the Loan perps
- [x] Verify that only the prep you selected has been added to the Loan perps
- [x] Verify that the prep you selected has been added to the Loan perps + all the siblings
- [x] Verify that only the prep you selected from the non conso has been added to the Loan perps + the prep from the conso COG and all its siblings

---

Adding COG record sets is working now.

In all the cases the preps are being grouped together only when appropriate 👍👍👍

**Review by sharadsw — APPROVED** [review]

**Review by acwhite211 — APPROVED** [review]

**Comment by grantfitzsimmons** [comment]
@CarolineDenis This PR does not follow the requirements as they were written– can you write a brief summary of what was skipped and why?

Re: https://github.com/specify/specify7/issues/5422

**Pull request merged**